hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/compact/CompactSerializationTest.java

```diff
@@ -430,6 +433,120 @@ public void testReflectiveSerializer_withFieldsWithOtherSerializationMechanisms_
         assertTrue(data.isCompact());
     }
 
+    @Test
+    public void testCompactWriterThrowsExceptionWhileWritingWhenSchemaDoesNotHaveThatField() {
+        FooBarSerializer serializer = spy(new FooBarSerializer());
+        FooBar fooBar = new FooBar("foo", 42L);
+
+        SchemaService schemaService = CompactTestUtil.createInMemorySchemaService();
+        CompactSerializationConfig compactSerializationConfig = new CompactSerializationConfig();
+        compactSerializationConfig.addSerializer(serializer);
+        SerializationService serializationService = new DefaultSerializationServiceBuilder()
+                .setSchemaService(schemaService)
+                .setConfig(new SerializationConfig().setCompactSerializationConfig(compactSerializationConfig))
+                .build();
```

Comment:
```suggestion
        SerializationService serializationService = createSerializationService(() -> serializer);
```
can be simplified to this

Reply:
👍🏻